File: hecho con python/app/controllers/estudiante_controller.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session
from typing import List
from app.config.database import get_db
from app.dtos.estudiante_dto import EstudianteCreateDTO, EstudianteUpdateDTO, EstudianteResponseDTO
from app.repositories.estudiante_repository import EstudianteRepository
from app.services.estudiante_service import EstudianteService

router = APIRouter(prefix="/estudiantes", tags=["Estudiantes"])

def get_estudiante_service(db: Session = Depends(get_db)) -> EstudianteService:
    repo = EstudianteRepository(db)
    return EstudianteService(repo)

@router.get("", response_model=List[EstudianteResponseDTO], summary="Obtener todos los estudiantes")
def obtener_todos_los_estudiantes(service: EstudianteService = Depends(get_estudiante_service)):
    return service.obtener_todos_los_estudiantes()

@router.get("/{id}", response_model=EstudianteResponseDTO, summary="Obtener estudiante por ID")
def obtener_estudiante_por_id(id: int, service: EstudianteService = Depends(get_estudiante_service)):
    return service.obtener_estudiante_por_id(id)

@router.get("/correo/{correo}", response_model=EstudianteResponseDTO, summary="Obtener estudiante por Correo")
def obtener_estudiante_por_correo(correo: str, service: EstudianteService = Depends(get_estudiante_service)):
    return service.obtener_estudiante_por_correo(correo)

@router.post("", response_model=EstudianteResponseDTO, status_code=status.HTTP_201_CREATED, summary="Crear estudiante")
def crear_estudiante(dto: EstudianteCreateDTO, service: EstudianteService = Depends(get_estudiante_service)):
    return service.crear_estudiante(dto)

@router.put("/{id}", response_model=EstudianteResponseDTO, summary="Actualizar estudiante")
def actualizar_estudiante(id: int, dto: EstudianteUpdateDTO, service: EstudianteService = Depends(get_estudiante_service)):
    return service.actualizar_estudiante(id, dto)

File: hecho con java/src/main/java/com/umanizales/notasya/controllers/EstudianteController.java
package com.umanizales.notasya.controllers;

import com.umanizales.notasya.dtos.EstudianteDTO;
import com.umanizales.notasya.services.EstudianteService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/estudiantes")
@Tag(name = "Estudiantes", description = "Endpoints para la gestión de estudiantes")
@CrossOrigin(origins = "*")
public class EstudianteController {

    private final EstudianteService service;

    public EstudianteController(EstudianteService service) {
        this.service = service;
    }

    @GetMapping
    @Operation(summary = "Obtener todos los estudiantes")
    public ResponseEntity<List<EstudianteDTO.Response>> obtenerTodos() {
        return ResponseEntity.ok(service.obtenerTodosLosEstudiantes());
    }

    @GetMapping("/{id}")
    @Operation(summary = "Obtener estudiante por ID")
    public ResponseEntity<EstudianteDTO.Response> obtenerPorId(@PathVariable Long id) {
        return ResponseEntity.ok(service.obtenerEstudiantePorId(id));
    }

    @GetMapping("/correo/{correo}")
    @Operation(summary = "Obtener estudiante por Correo")
    public ResponseEntity<EstudianteDTO.Response> obtenerPorCorreo(@PathVariable String correo) {
        return ResponseEntity.ok(service.obtenerEstudiantePorCorreo(correo));
    }

    @PostMapping
    @Operation(summary = "Crear nuevo estudiante")
    public ResponseEntity<EstudianteDTO.Response> crear(@Valid @RequestBody EstudianteDTO.Create dto) {
        return ResponseEntity.status(HttpStatus.CREATED).body(service.crearEstudiante(dto));
    }

    @PutMapping("/{id}")
    @Operation(summary = "Actualizar estudiante existente")
    public ResponseEntity<EstudianteDTO.Response> actualizar(@PathVariable Long id, @Valid @RequestBody EstudianteDTO.Update dto) {
        return ResponseEntity.ok(service.actualizarEstudiante(id, dto));
    }
}
